install_script_op_worker.sh

```diff
@@ -0,0 +1,553 @@
+support_email="support@datadoghq.com"
+variant="observability-pipelines-worker"
+
+# DATADOG_APT_KEY_CURRENT.public always contains key used to sign current
+# repodata and newly released packages
+# DATADOG_APT_KEY_F14F620E.public expires in 2032
+# DATADOG_APT_KEY_C0962C7D.public expires in 2028
+APT_GPG_KEYS=("DATADOG_APT_KEY_CURRENT.public" "DATADOG_APT_KEY_C0962C7D.public" "DATADOG_APT_KEY_F14F620E.public")
+
+# DATADOG_RPM_KEY_CURRENT.public always contains key used to sign current
+# repodata and newly released packages
+# DATADOG_RPM_KEY_FD4BF915.public expires in 2024
+# DATADOG_RPM_KEY_B01082D3.public expires in 2028
+RPM_GPG_KEYS=("DATADOG_RPM_KEY_CURRENT.public" "DATADOG_RPM_KEY_B01082D3.public" "DATADOG_RPM_KEY_FD4BF915.public")
+
+# Set up a named pipe for logging
+npipe=/tmp/$$.tmp
+mknod $npipe p
+
+# Log all output to a log for error checking
+tee <$npipe $logfile &
+exec 1>&-
+exec 1>$npipe 2>&1
+trap 'rm -f $npipe' EXIT
+
+function fallback_msg(){
+  printf "
+If you are still having problems, please send an email to $support_email
+with the contents of $logfile and any information you think would be
+useful and we will do our very best to help you solve your problem.\n"
+}
+
+function report(){
+  if curl -f -s \
+    --data-urlencode "os=${OS}" \
+    --data-urlencode "version=${worker_major_version}" \
+    --data-urlencode "log=$(cat $logfile)" \
+    --data-urlencode "email=${email}" \
+    --data-urlencode "apikey=${apikey}" \
+    --data-urlencode "variant=${variant}" \
+    "$report_failure_url"; then
+   printf "A notification has been sent to Datadog with the contents of $logfile\n"
+  else
+    printf "Unable to send the notification (curl v7.18 or newer is required)"
+  fi
+}
+
+function on_read_error() {
+  printf "Timed out or input EOF reached, assuming 'No'\n"
+  yn="n"
+}
+
+function get_email() {
+  emaillocalpart='^[a-zA-Z0-9][a-zA-Z0-9._%+-]{0,63}'
+  hostnamepart='[a-zA-Z0-9.-]+\.[a-zA-Z]+'
+  email_regex="$emaillocalpart@$hostnamepart"
+  cntr=0
+  until [[ "$cntr" -eq 3 ]]
+  do
+      read -p "Enter an email address so we can follow up: " -r email
+      if [[ "$email" =~ $email_regex ]]; then
+        isEmailValid=true
+        break
+      else
+        ((cntr=cntr+1))
+        echo -e "\033[33m($cntr/3) Email address invalid: $email\033[0m\n"
+      fi
+  done
+}
+
+function on_error() {
+    if [ -z "${ERROR_MESSAGE}" ] ; then
+      # Save the few lines of the log file for telemetry if the error message is blank
+      SAVED_ERROR_MESSAGE=$(tail -n 3 $logfile)
+    fi
+
+    printf "\033[31m$ERROR_MESSAGE
+It looks like you hit an issue when trying to install the $nice_flavor.
+
+Troubleshooting and basic usage information for the $nice_flavor are available at:
+
+    https://docs.datadoghq.com/observability_pipelines/\n\033[0m\n"
+
+    ERROR_MESSAGE=$SAVED_ERROR_MESSAGE
+
+    if ! tty -s; then
+      fallback_msg
+      exit 1;
+    fi
+
+    if [ "$site" == "ddog-gov.com" ]; then
```

Comment:
Would we want to fail that inside the install script, I can imagine that being forgotten if we ever do support Gov Cloud. It'd be easy enough to do that check and exit the script early if desired.

cc @barieom @austinchen15

Reply:
I'm in favor of doing it in the script! I don't think it'll be forgotten if we do ever support Gov cloud. 

Besides, if we do, it will error out quickly and be a relatively easy error to fix as long as we leave a breadcrumb comment for future engineers to disable this logic upon Gov Cloud enablement.

Reply:
Added.